src/core/src/GslMatrix.C

```diff
@@ -432,6 +441,50 @@ GslMatrix::chol()
   return iRC;
 }
 
+void
+GslMatrix::cholSolve(const GslVector & rhs, GslVector & sol) const
+{
+  queso_require_equal_to_msg(this->numCols(), rhs.sizeLocal(), "matrix and rhs have incompatible sizes");
+  queso_require_equal_to_msg(sol.sizeLocal(), rhs.sizeLocal(), "solution and rhs have incompatible sizes");
+
+  int iRC;
+  if (m_chol == NULL) {
+    m_chol = gsl_matrix_calloc(this->numRowsLocal(), this->numCols());
+    queso_require_msg(m_chol, "gsl_matrix_calloc() failed");
+
+    iRC = gsl_matrix_memcpy(m_chol, m_mat);
+    queso_require_msg(!(iRC), "gsl_matrix_memcpy() failed");
+
+    gsl_error_handler_t * oldHandler;
+    oldHandler = gsl_set_error_handler_off();
+
+    iRC = gsl_linalg_cholesky_decomp(m_chol);
```

Comment:
The code here could be shared with the chol() function, though the behavior and error handling are slightly different.